### PR TITLE
Migrate Parameter Parsing: moved configuration of `EnzoInitiailCloud` into constructor

### PR DESCRIPTION
**Pull request by mabruzzo**
### Pull request summary

Followup from #349: moved parameter parsing for `EnzoInitiailCloud` from `EnzoConfig` into constructor.

**Aside:** This sort of migration is the sort of thing that needs to be done for a lot more examples. This is a great first-time-contribution for new developers. I'm going to create an Issue summarizing the process and that links to this PR as an example (in reality, I will probably link to the particular commit that makes this change).

### CODE REVIEW

**Review by jwise77 — APPROVED** [review]

File: src/Enzo/initial/EnzoInitialCloud.hpp [hunk 1]
```diff
@@ -23,64 +23,80 @@ class EnzoInitialCloud : public Initial {
...
+      cloud_center_x_(p.value_float("cloud_center_x",0.0)),
+      cloud_center_y_(p.value_float("cloud_center_y",0.0)),
+      cloud_center_z_(p.value_float("cloud_center_z",0.0)),
+      density_cloud_(p.value_float("cloud_density",0.0)),
+      density_wind_(p.value_float("wind_density",0.0)),
+      etot_wind_(p.value_float("wind_total_energy",0.0)),
+      eint_wind_(p.value_float("wind_internal_energy",0.0)),
+      velocity_wind_(p.value_float("wind_velocity",0.0)),
+      metal_mass_frac_(p.value_float("metal_mass_fraction",0.0)),
+      initialize_uniform_bfield_(false),
```
Comment (clairekope) [inline src/Enzo/initial/EnzoInitialCloud.hpp:39]: What is the purpose of this parameter if `initialize_uniform_bfield_` is set on lines 61-72 based on the presence of a `uniform_bfield` parameter list?

**Review by clairekope — APPROVED** [review]
Small question about the need of a parameter but the port looks good

File: src/Enzo/initial/EnzoInitialCloud.hpp (hunk 1 above)
Comment (mabruzzo) [reply src/Enzo/initial/EnzoInitialCloud.hpp:39]: good point

Comment (mabruzzo) [reply src/Enzo/initial/EnzoInitialCloud.hpp]: fixed

**Pull request merged**